turbopack/crates/turbo-tasks/src/backend.rs

```diff
@@ -213,9 +213,7 @@ mod ser {
                 unreachable!();
             };
             let mut state = serializer.serialize_seq(Some(2))?;
-            // It would be a little tricky but we _could_ store the function id on the native_fn so
-            // that instead of a hashmap lookup this would be a memory access on the NativeFn
-            state.serialize_element(&registry::get_function_id(native_fn))?;
```

Comment:
That fixes a bug. Storing the function id is not correct as it is not deterministic.

Reply:
what was the bug? we would pass the function id here, but then `serialize_element` would select the `Serialize` implementation of `FunctionId` so i think it was the same. tbh i am a little confused about this serialization api